### PR TITLE
Fix active element detection in shadow root

### PR DESCRIPTION
**Pull request by alexanderjulmer**
As mentioned in #24, when used inside a shadowRoot, the detection whether the input textarea is active, fails. This is because `document.activeElement` only returns the CustomElement but does not check inside its shadowRoot. This PR changes this by extending the check to also look inside the shadowRoot.

The provided test also adds a dummy CustomElement (`WrapperComponent`) to test usage inside a shadowRoot.

### CODE REVIEW

File: test/text-expander-element-test.js [hunk 1]
```diff
@@ -204,6 +206,31 @@ describe('text-expander element', function () {
       assert.equal('step 1 #step 2 #step 3', text)
     })
   })
+
+  describe('use in customElement', function () {
```
Comment (keithamus) [inline test/text-expander-element-test.js:210]: Minor nit: 
```suggestion
  describe('use inside a ShadowDOM', function () {
```

**Review by keithamus — APPROVED** [review]
This LGTM :+1:

Let's get a second :eyes: from someone on @github/ui-frameworks

**Review by koddsson — APPROVED** [review]
This looks really good to me. The tests make this really easy to review.

File: test/text-expander-element-test.js [hunk 2]
```diff
@@ -204,6 +206,31 @@ describe('text-expander element', function () {
       assert.equal('step 1 #step 2 #step 3', text)
     })
   })
+
+  describe('use inside a ShadowDOM', function () {
+    this.beforeAll(function () {
```
[the remark this answers is not in the document]
Comment (alexanderjulmer) [reply test/text-expander-element-test.js]: I don't think it is. In fact, [the docs](https://mochajs.org/#hooks) use `before()` (which then calls `beforeAll()`; [source](https://mochajs.org/api/interfaces_common.js.html#line59)). Anyway, changed it to `before()`.

**Review by dgreif — APPROVED** [review]

**Pull request merged**